undefined reference to `__stack_chk_guard' while compiling nvim 0.4.2

I am trying to compile neovim v0.4.2 on a linux x86_64 with a kernel 3.18.0-19733-g972ff9adcd38. It is a chromebook Asus C302CA. I am using chromebrew and I am updating the neovim package.
I am facing this error after solving #11198 
`make distclean && make` produces this error :
```
-- Build files have been written to: /usr/local/tmp/crew/v0.4.2.tar.gz.dir/neovim-0.4.2/build
touch build/.ran-cmake
ninja -C build
ninja: Entering directory `build'
[136/310] Linking C executable bin/shell-test
FAILED: bin/shell-test 
: && /usr/local/bin/cc -g  -Wl,--no-undefined   -rdynamic test/functional/fixtures/CMakeFiles/shell-test.dir/shell-test.c.o  -o bin/shell-test   && :
/usr/local/bin/ld: test/functional/fixtures/CMakeFiles/shell-test.dir/shell-test.c.o: in function `main':
/usr/local/tmp/crew/v0.4.2.tar.gz.dir/neovim-0.4.2/build/../test/functional/fixtures/shell-test.c:48: undefined reference to `__stack_chk_guard'
/usr/local/bin/ld: /usr/local/tmp/crew/v0.4.2.tar.gz.dir/neovim-0.4.2/build/../test/functional/fixtures/shell-test.c:138: undefined reference to `__stack_chk_guard'
collect2: error: ld returned 1 exit status
[138/310] Generating auto/unicode_tables.generated.h
ninja: build stopped: subcommand failed.
make: *** [Makefile:86: nvim] Error 1
```

> If you pass in `CFLAGS` it gets used always. This might be good as a workaround, but we should figure out a way to not require it.

Fully agree. This hack will be perfect for my very actual need, but if we find a way to solve this cleaner, this would be great.

> This is really strange:
> > check_c_compiler_flag(-fstack-protector-all HAS_FSTACK_PROTECTOR_ALL_FLAG)
> > Performing Test HAS_FSTACK_PROTECTOR_ALL_FLAG - Failure 
> For me it also succeeds.
> This might indicate that there are flags already, so that this fails.
> > and -fstack-protector-all is found everywhere twice in the build.ninja file.

Maybe I wasn't clear enough, but it seems to act as if `-fstack-protector-strong` is set by default. An interesting thing is that it is the default value when `--enable-default-ssp` is set as found in `defaults.h` in gcc sources.
 ```
/* Default value for flag_stack_protect when flag_stack_protect is initialized to -1:                                                                                                          
    --enable-default-ssp: Default flag_stack_protect to -fstack-protector-strong.                                                                                                               
    --disable-default-ssp: Default flag_stack_protect to 0.                                                                                                                                     
*/                                                                                                                                                                                            
#ifdef ENABLE_DEFAULT_SSP                                                                                                                                                                      
# ifndef DEFAULT_FLAG_SSP                                                                                                                                                                      
#  define DEFAULT_FLAG_SSP 3                                                                                                                                                                   
# endif                                                                                                                                                                                        
#else                                                                                                                                                                                          
# define DEFAULT_FLAG_SSP 0                                                                                                                                                                    
#endif  
```
When no argument is passed to make, --fstak-protector-strong is found in build.ninja. When passing an argument, it is duplicated.

> Do you have `-fstack-protector-all` already once there with just `make distclean && make cmake`?
> This would indicate that this is used already, and might cause the issue.

with just `make distclean && make cmake`, `-fstack-protector-strong` is found once in `build.ninja` lines amd in `compile_commands.json`

> Do you (or chromebrew) set CLAGS or LDFLAGS already somehow? (except from when passing it explicitly like mentioned above, of coursE)
> Try `message(FATAL_ERROR "CFLAGS=$ENV{CFLAGS} LDFLAGS=$ENV{LDFLAGS}")` at the top of `CMakeLists.txt` and then `rm build -rf && make cmake`.

This shows nothing relevant here unfortunately... 
```
cd build && cmake -G 'Ninja' -DCMAKE_BUILD_TYPE=Debug  /usr/local/tmp/crew/v0.4.2.tar.gz.dir/neovim-0.4.2
CMake Error at CMakeLists.txt:7 (message):
  CFLAGS= LDFLAGS=
```

From the last log:
```
-- Performing Test HAS_FSTACK_PROTECTOR_STRONG_FLAG
-- Performing Test HAS_FSTACK_PROTECTOR_STRONG_FLAG - Success
-- Performing Test HAS_FSTACK_PROTECTOR_ALL_FLAG
-- Performing Test HAS_FSTACK_PROTECTOR_ALL_FLAG - Failed
-- Performing Test HAS_FSTACK_PROTECTOR_FLAG
-- Performing Test HAS_FSTACK_PROTECTOR_FLAG - Success
…
[135/310] /usr/local/bin/cc -DINCLUDE_GENERATED_DECLARATIONS -DNVIM_MSGPACK_HAS_FLOAT32 -DNVIM_UNIBI_HAS_VAR_FROM -D_GNU_SOURCE -Iconfig -I../src -isystem ../.deps/usr/include -g   -Wall -Wextra -pedantic -Wno-unused-parameter -Wstrict-prototypes -std=gnu99 -Wshadow -Wconversion -Wmissing-prototypes -Wimplicit-fallthrough -Wvla -fstack-protector-strong -fdiagnostics-color=always -MD -MT test/functional/fixtures/CMakeFiles/shell-test.dir/shell-test.c.o -MF test/functional/fixtures/CMakeFiles/shell-test.dir/shell-test.c.o.d -o test/functional/fixtures/CMakeFiles/shell-test.dir/shell-test.c.o   -c ../test/functional/fixtures/shell-test.c
[136/310] : && /usr/local/bin/cc -g  -Wl,--no-undefined   -rdynamic test/functional/fixtures/CMakeFiles/shell-test.dir/shell-test.c.o  -o bin/shell-test   && :
FAILED: bin/shell-test
: && /usr/local/bin/cc -g  -Wl,--no-undefined   -rdynamic test/functional/fixtures/CMakeFiles/shell-test.dir/shell-test.c.o  -o bin/shell-test   && :
/usr/local/bin/ld: test/functional/fixtures/CMakeFiles/shell-test.dir/shell-test.c.o: in function `main':
/usr/local/tmp/crew/v0.4.2.tar.gz.dir/neovim-0.4.2/build/../test/functional/fixtures/shell-test.c:48: undefined reference to `__stack_chk_guard'
/usr/local/bin/ld: /usr/local/tmp/crew/v0.4.2.tar.gz.dir/neovim-0.4.2/build/../test/functional/fixtures/shell-test.c:138: undefined reference to `__stack_chk_guard'
collect2: error: ld returned 1 exit status
```

Note that you have your custom addition still there (which might be OK), but
it is still puzzling while it fails, when it makes it work when forced via
`CFLAGS`:

> -- Performing Test HAS_FSTACK_PROTECTOR_ALL_FLAG
> -- Performing Test HAS_FSTACK_PROTECTOR_ALL_FLAG - Failed

Unfortunately I am out of ideas for now.

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -303,8 +303,10 @@ if(UNIX)
 
   if(HAS_FSTACK_PROTECTOR_STRONG_FLAG)
     add_compile_options(-fstack-protector-strong)
+    link_libraries(-fstack-protector-strong)
   elseif(HAS_FSTACK_PROTECTOR_FLAG)
     add_compile_options(-fstack-protector --param ssp-buffer-size=4)
+    link_libraries(-fstack-protector --param ssp-buffer-size=4)
   endif()
 endif()
 

